src/FunctionalExtras/Data/Either.cs

```diff
@@ -0,0 +1,205 @@
+
+    /// <summary>
+    /// Curried implementation of <see cref="Either.FromRight{T, U}(U, IEither{T, U})"/>.
+    /// </summary>
+    public static Func<IEither<T, U>, U> FromRight<T, U>(U defaultValue) => either => FromRight(defaultValue, either);
+
+    /// <summary>
+    /// Extracts the value of a <code>Right</code>; otherwise, returns the <code>defaultValue</code>.
+    /// </summary>
+    /// <returns>The underlying right value or default.</returns>
+    /// <param name="defaultValue">Value used if the <code>either</code> is not a <code>Right</code>.</param>
+    /// <param name="either">The <see cref="Either"/>.</param>
+    /// <typeparam name="T">The underlying left type parameter.</typeparam>
+    /// <typeparam name="U">The underlying right type parameter.</typeparam>
+    public static U FromRight<T, U>(U defaultValue, IEither<T, U> either) => either is Right<T, U> right
+      ? right._value
+      : defaultValue;
+
+    /// <summary>
+    /// Determines whether or not the instance is a <code>Left</code>.
+    /// </summary>
+    /// <returns><code>true</code> for a <code>Left</code>; otherwise, <code>false</code>.</returns>
+    /// <param name="either">The <see cref="Either"/>.</param>
+    /// <typeparam name="T">The underlying left type parameter.</typeparam>
+    /// <typeparam name="U">The underlying right type parameter.</typeparam>
+    public static bool IsLeft<T, U>(IEither<T, U> either) => either.IsLeft();
+
+    /// <summary>
+    /// Determines whether or not the instance is a <code>Right</code>.
+    /// </summary>
+    /// <returns><code>true</code> for a <code>Right</code>; otherwise, <code>false</code>.</returns>
+    /// <param name="either">The <see cref="Either"/>.</param>
+    /// <typeparam name="T">The underlying left type parameter.</typeparam>
+    /// <typeparam name="U">The underlying right type parameter.</typeparam>
+    public static bool IsRight<T, U>(IEither<T, U> either) => either.IsRight();
+
+    /// <summary>
+    /// Creates a <code>Left</code> from an arbitrary <paramref name="value"/>.
+    /// </summary>
+    /// <returns>A <code>Left</code> of the value.</returns>
+    /// <param name="value">The value.</param>
+    /// <typeparam name="T">The underlying left type parameter.</typeparam>
+    /// <typeparam name="U">The underlying right type parameter.</typeparam>
+    public static IEither<T, U> Left<T, U>(T value) => new Left<T, U>(value);
+
+    /// <summary>
+    /// Extracts from a list of <see cref="Either"/> all of the <code>Left</code> elements in extracted order.
+    /// </summary>
+    /// <returns>The underlying <code>Left</code> values.</returns>
+    /// <param name="list">The list of <see cref="Either"/>.</param>
+    /// <typeparam name="T">The underlying left type parameter.</typeparam>
+    /// <typeparam name="U">The underlying right type parameter.</typeparam>
+    public static List<T> Lefts<T, U>(List<IEither<T, U>> list) => (list ?? new List<IEither<T, U>>())
```

Comment:
The argument here probably ought to be `IEnumerable<IEither<T, U>>`, instead of `List<IEither<T, U>>`.  That makes it work with any enumerable type, but it _does_ raise the specter of needing to make sure we don't enumerate things multiple times.  Whether or not the return value should be `IEnumerable` is another question that I haven't fully considered yet.

Reply:
Bear in mind that this is not an enumeration of database rows, but of established list of `Either`. These have already been processed once since most DB connections don't return an enumeration of _this_ implementation of `Either`.

Reply:
Fair point on the database rows, but you're still limiting it to only `List`.  If someone was using `LinkedList`, or some hand-rolled enumerable tree, this wouldn't work for them.  I think the general wisdom is to take in `IEnumerable` unless you can 100% guarantee you'll never have more than `List`.

Reply:
Applied the `IEnumerable` changes to all the things I could think of. You'll want to give it a double check that I didn't miss it somewhere.